### ISSUE TITLE
Update download unsuccessful

### ISSUE THREAD

**Issue by caindrac** [FezVrasta/ark-server-tools#671]
Running command 'update' for instance 'main'
2016-11-28 18:53:14: checking for update; PID: 659
Update download unsuccessful

I am getting this error when I try to update my server. It is currently up-to date but I just don't know what is going on.

**Comment by NexGenHD** [comment]
I have exactly the same problem, Any solution?

**Comment by NexGenHD** [comment]
I specify a little more:
2:~$ arkmanager checkupdate
Running command 'checkupdate' for instance 'main'
Querying Steam database for latest version...
Current version:  1484033
Available version:  1484056
Your server needs to be restarted in order to receive the latest update.
Run "arkmanager update" to do so

And when I run it:
2:~$ arkmanager update
Running command 'update' for instance 'main'
2016-11-28 20:49:33: checking for update; PID: 9354
Downloading ARK update ... Update download unsuccessful

**Comment by CaspyBXL** [comment]
Same here, any fix ? Downloading ARK update ... Update download unsuccessful

**Comment by Gumballz** [comment]
Same

**Comment by CaspyBXL** [comment]
Results and temp fix : 

Running command 'status' for instance 'test2'
 Server version:   1484033

Weird thing is the version on steam says this is version (v252.22)

I have been able to update the server by just doing the following manual commands : 

arkmanager shutdown --warn --saveworld @test2
Once server stopped ... 
arkmanager update @test2
after update done

arkmanager start @test2

Server Name: CLUSTER TEST 2 - (v252.22)
Players: 0 / 30
 Server version:   1484056

So it seems both versions 1484033 & 1484056 are (v252.22)

So for now i uses the manual commands to get servers offline first with a save/backup, then updating it by manual command, then restarting the server.

Greetings,

**Comment by Gumballz** [comment]
quick workaround until fixed....

Login to steamcmd 
```
./steamcmd/steamcmd.sh
login anonymous
force_install_dir ARK-Staging
app_update 376030 validate
exit
Arkmanager update @instance
```

**Comment by klightspeed** [comment]
The same thing can be done using `arkmanager update --validate`

**Comment by Lihis** [comment]
Yeah, @klightspeed was faster than me, `arkmanager update --validate` did the trick.

**Comment by Gumballz** [comment]
what the difference between --force and --validate then ?

Previously i was using --force (thinking it would force an update whether one was required or not)

**Comment by klightspeed** [comment]
`--force` simply forces the Steam update without checking if the version has changed.

`--validate` passes the `validate` option to SteamCMD

**Comment by klightspeed** [comment]
v1.6.11 should fix the `Update download unsuccessful` issue for those not using a staging directory.

If you are stuck at build 1484033, you will need to use the `--validate` option to update to build 1484056.  It seems the real issue is that Wildcard updated the build ID without actually changing any files (the manifest did not change between 1484033 and 1484056).

**Comment by Gumballz** [comment]
Thanx for the speedy work @klightspeed 👍 

If only WC was half as efficient /sigh

**Comment by rob-bb** [comment]
Even With the --validate option I am stuck at 1484033

**Comment by klightspeed** [comment]
v1.6.12 should fix the update loop for those not using a staging directory.

Wildcard pushed a new build for both server and client (both have a buildid of 1484056).  However, the server build contained no changes (and thus has the same manifest).  It would appear that SteamCMD only checks the manifests and not the build ID, and because the manifests hadn't changed it wasn't updating the build ID.

v1.6.12 checks if the manifests have changed, and if they haven't then it won't perform an update unless `--force` or `--validate` are used.

**Comment by CaspyBXL** [comment]
Just a stupid question, do i have to take down my servers to update to v1.6.12 ? Never did it with servers online ... and my peeps just got 12 reboots yesterday because i was not there when the loop started and i have a Cron running every hour xD

**Comment by rob-bb** [comment]
I update mine all the time while the server is running and haven't had a problem.

Mine was updating every 30 mins :(

**Comment by CaspyBXL** [comment]
All Good Rob, All Good !

Tools - ok
Servers - ok
Sleep - not ok

Good Night !

**Comment by rob-bb** [comment]
@klightspeed Thanks for all your time and effort with this, it is much appreciated.  I don't want to imagine running an ark server without using your tools.  Thank you.

**Comment by BigMikJ** [comment]
Need to reopen this one.... been happening ever since the last 4 updates. have to manually do it each time now. Please assist with a resolution to this. Thanks ahead of time!

[root@NUKEAPP02 ~]# arkmanager update --verbose
Running command 'update' for instance 'ragnarok'
Checking for update; PID: 93926
Downloading ARK updateExecuting /app_data/home/steam/Steam/steamcmd.sh +@NoPromptForPassword 1 +login anonymous +force_install_dir /app_data/ark/staging +app_update 376030 +quit
Redirecting stderr to '/app_data/home/steam/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation
-- type 'quit' to exit --
Loading Steam API...OK.
"@NoPromptForPassword" = "1"

Connecting anonymously to Steam Public...Logged in OK
Waiting for user info...OK
ERROR! Timed out waiting for AppInfo update.
Update download unsuccessful
[root@NUKEAPP02 ~]#

**Comment by BigMikJ** [comment]
I've cleared out the "appcache" folder and also emptied the "Staging" directory. Still no help :(

**Comment by klightspeed** [comment]
@BigMikJ Please see #1043
